### ISSUE TITLE
Skeleton updates made HARDTOSHOOT erroneous for non-skeletal monsters

### ISSUE THREAD

**Issue by Barhandar** [CleverRaven/Cataclysm-DDA#27786]
**Is your feature request related to a problem? Please describe.**  
New implementation of HARDTOSHOOT, namely, reducing effective size for ranged calculations, follows new skeleton lore (bone-covered shrinkwrapped zombies, as opposed to literally walking endoskeletons); however, that completely ignored that the same tag applies to monsters that _are_ actually likely to get missed entirely even on a successful shot, such as bats (twitchy; and pretty sure they're already MS_TINY and so the flag doesn't even work on them), wraiths (described as chaotically changing gigantic shadows, so character _cannot_ pinpoint where the actually damage-able area is) and shadows, holograms, and smoke zombies.

**Describe the solution you'd like**  
Restore previous behavior of HARDTOSHOOT (chance to miss entirely unless using WIDE projectiles), add a separate flag with current behavior (reduced effective size) and apply it to skeletons; or vice versa (keep HARDTOSHOOT but add new flag with old behavior).
Monsters that are extremely thin, twitchy or have obscured/invisible "body" should be reliably hit with area-hitting ammo, not with "getting a bit better at aiming".

**Describe alternatives you've considered**  
None; I consider this change to HARDTOSHOOT a breakage.

**Comment by nexusmrsep** [comment]
It works as intended. If you shoot a bat in the gut it's dead meat, it doesn't phase out second before the bullet hits it. Wraith isn't a ghost that would pass bullets through it's body, same for other mentioned monsters. They are harder to hit because of their visual traits, and it is simulated by reducing their size for the purpose of hitting calculation, but they never were meant to be ethereal beings, so sticking a gun barrel to the dome of their heads from one square away is as effective as it would be in case of any other monster.

**Comment by Barhandar** [comment]
>If you shoot a bat in the gut it's dead meat

The projectiles aren't hitscan (except for lasers), the bat might jerk away or simply move in unpredictable manner - which is what previous mechanic, miss chance scaling down with better skills, simulated.
>and it is simulated by reducing their size for the purpose of hitting calculation

See above about "some of these are already minimal size and hence it doesn't work".
>Wraith isn't a ghost that would pass bullets through it's body, same for other mentioned monsters.

The character doesn't _know_ where said body is. The only legible part is the eyes, and these shift around the same as the rest of it. Shadows, from what I can see, _literally are ghosts_ that ignore most things and vanish in the sunlight. Stuff like Amigara Horrors doesn't even have body wide and slow enough (they're two story tall noodles) to do reliable hits on.
>same for other mentioned monsters

Holograms aren't even solid in the first place.
>so sticking a gun barrel to the dome of their heads from one square away

That is not simulated by current mechanic either; in fact, it's _worse_ at simulating that than "scale miss chance with distance". Also, good luck doing that to a bat mid-flight.

**Comment by kevingranade** [comment]
Bats and smoke zombies are definitely working as intended.  I'm very inclined to say that wraiths and shadows are as well.  I don't know why holograms are even *possible* to shoot given the description.

More importantly, "you have a 50% miss chance no matter how good you are" is a terrible game mechanic.  There's a reason I got rid of it.

**Comment by Barhandar** [comment]
>More importantly, "you have a 50% miss chance no matter how good you are" is a terrible game mechanic. There's a reason I got rid of it.

"Unless you're using buckshot or a flamethrower". Which put value on actually preparing for specific things, however, I do agree that 50% minimum miss chance is a bit high and that it should've been something like minimum 0% (at ridiculously high skill), maximum 50% (at skill levels where you'd be hitting target consistently without it) instead. 

My point boils down to "the current HARDTOSHOOT is not drastic enough and, unless SHOT and WIDE (and logically also LASER, because those are effective hitscan) projectiles ignoring it was kept, invalidates some of player preparation".

**Comment by kevingranade** [comment]
Neat we're getting somewhere now. I think we're basically in agreement that literally bringing back the "50% miss chance" isn't a good option, so let's poke at the use cases...

I'm not sure about HARDTOSHOOT not being drastic enough, it's pretty drastic.  I'll have to look at some numbers on that one.  However just based on what I remember of SHOT, WIDE and SPREAD type handling, I'm very inclined to believe that it doesn't do what you would expect.

e.g. Shooting at the bat in particular is exactly the kind of thing birdshot is intended for (not buckshot, but whatever).  However I already have an outline of how I want to deal with that, which is [here](https://discourse.cataclysmdda.org/t/breaking-the-more-accurate-weapons-are-always-better-disfunction/14067?u=kevin.granade).  Tl;dr, things moving around erratically produce an effect that interferes with the ability to aim at them (dodge). Bats would have a quite high score at this kind of thing, making them extremely difficult to hit with a pistol or rifle, but due to birdshot spread, a character should be able to aim in the general vicinity of a fluttering bat and expect a hit. So that's a two-part solution, add the dodge effect and apply it consistently, and fix the SHOT type effect to better model how a shotgun actually works.

However, that particular effect doesn't have much impact on any of the larger HARDTOSHOOT targets, the spread for SHOT just isn't that wide in absolute terms.

I will 100% buy that SPREAD and WIDE should be using a much different mechanism where they actually hit things in a significantly sized area, basically making it "impossible to miss" in that it "hits everything in the target square(s).  So that's another fix.

Wraiths and smoke zombies potentially have some effect that works similarly to dodge in that it restricts how well you are able to draw a bead on them, but via obscuration rather than movement.  I'm inclined to just call that dodge.

IDK what I want to do with the actually incoporeal stuff.  If it's incorporeal, no amount of "more bullets" should make any difference. Handing out a pass is a game-ism to avoid having that one kind of effect dominate play, but I don't agree with that, if it's incoporeal, you need something that can deal with incoporeal things, full stop.

**Comment by Barhandar** [comment]
>I'm not sure about HARDTOSHOOT not being drastic enough, it's pretty drastic.

Skeletal juggernauts  (and wraiths/fungal towers) are HUGE, HARDTOSHOOT drops them to LARGE; by the time survivor sees these, barring necromancers, they'll be able to reliably hit MEDIUM enemies (regular zombies), much less a LARGE one. Just checked and it does in fact do nothing to bats, they're already TINY; so it currently only affects enemies that are normally MEDIUM, that is, skeletons, howling waifs, shadows, and similar.
> Tl;dr, things moving around erratically produce an effect that interferes with the ability to aim at them (dodge). Bats would have a quite high score at this kind of thing, making them extremely difficult to hit with a pistol or rifle, but due to birdshot spread, a character should be able to aim in the general vicinity of a fluttering bat and expect a hit.

Yeah, this.
>However, that particular effect doesn't have much impact on any of the larger HARDTOSHOOT targets, the spread for SHOT just isn't that wide in absolute terms.

Depends on the target; character can be expected to not miss stuff like skeletons and juggernauts, moreso that they're slowpokes, with regular weapons as well (they're shrinkwrapped, not hollow), but wraiths are questionable - SHOT still would be much better than single bullet for hitting a target you cannot see well that _also_ moves chaotically, as per description. Fungal towers depend on how dense their tendrils actually are - if they're not very, SHOT would be better at partial hits, and liquids/lasers would outright ignore tendrils. Or the player could just shoot the solid cap.
>I will 100% buy that SPREAD and WIDE should be using a much different mechanism where they actually hit things in a significantly sized area, basically making it "impossible to miss" in that it "hits everything in the target square(s). So that's another fix.

In that case precision and range would instead determine _how much_ of the projectile load (liquid area, shot pellets) hit the target, that is,  damage percentage.
>IDK what I want to do with the actually incoporeal stuff. If it's incorporeal, no amount of "more bullets" should make any difference.

Depends on how incorporeal it is. For example, its "grasp on reality" - since it needs to attack you successfully or it might as well be a hallucination, because the game only tracks physical damage for now - might be able to handle one or two bullets fine, but shot and flame, not so much. However, the logical consequence is that you'd also be able to hurt such things by yelling at them really loudly (i.e. waving the air they're inside of). I think that's funny and an acceptable way to damage incorporeal things.

**Comment by kevingranade** [comment]
Stalled discussion, basically brings up some points that are already planned and documented elsewhere.
